projects/state of rust adoption survey/2025/proposed_questions.md

```diff
@@ -0,0 +1,262 @@
+ 
+- [ ] Memory safety
+- [ ] Thread safety
+- [ ] Speed/performance
+- [ ] Tooling (cargo, crates.io, clippy, etc)
+- [ ] Language features (type system, matching, etc)
+- [ ] Safe by default language design
+- [ ] Zero cost abstractions
+- [ ] Specific library/crate/framework
+- [ ] Hiring advantages (interest in Rust/quality of Rust engineers)
+- [ ] Regulatory requirement
+- [ ] Cybersecurity advantages
+- [ ] Other (text entry)
+
+What are the primary blockers to using Rust in your safety critical role?
+ *Select multiple*
+
+- [ ] Lack of Tooling
+- [ ] Standards
+- [ ] Hiring engineers
+- [ ] Hardware support
+- [ ] Ecosystem (libraries, drivers, etc)
+- [ ] Regulatory hurdles
+- [ ] Legacy Codebase
+- [ ] Training
+- [ ] Other (text entry)
+
+How would you be interested in using Rust in your safety-critical role?
+*Select multiple*
+
+- [ ] Starting a new project in Rust
+- [ ] Rewriting an existing project in Rust
+- [ ] Integrating Rust with an existing codebase
+- [ ] Integrating Rust components into an existing system
+- [ ] Other (text entry)
+
+### Not Interested in Using Rust
+
+What are the primary disadvantages to using Rust in your safety critical role?
+ *Select multiple*
+
+- [ ] Doesn't offer any/enough advantage over existing languages
+- [ ] Too difficult to learn
+- [ ] Doesn't support my hardware target(s)
+- [ ] Not certifiable
+- [ ] Not enough hirable Rust engineers
+- [ ] Doesn't integrate with existing codebase
+- [ ] Doesn't integrate with existing process/tooling/build system
+- [ ] Software supply chain concern (open-source)
+- [ ] Don't know enough about it
+- [ ] Other (text entry)
+
+## Tooling
+
+What types of safety critical code tools does your work require?
+ *Select multiple*
+
+- [ ] Certified compiler
+- [ ] Formal verification
+- [ ] Code coverage analysis
+- [ ] Code-requirements traceability
+- [ ] Static Analysis
+- [ ] Code metrics - cyclomatic complexity
+- [ ] Linting tools (naming conventions, style guides, etc)
+- [ ] Automated testing (fuzz, prop-based, etc)
+- [ ] Other (text entry)
+
+## Standards/Regulation
+
+What standard(s) do you work with in your safety critical work?
+ *Select multiple*
+
+- [ ] ISO-26262
+- [ ] DO-178
+- [ ] IEC-61508
+- [ ] IEC-62304
+- [ ] Other (text entry)
+
+### Standards Levels
+
+The following questions are only shown to the respondent if they checked that they work with that specific standard.
+
+What levels of ISO-26262 do you work with?
+
+- [ ] QM
+- [ ] ASIL-A
+- [ ] ASIL-B
+- [ ] ASIL-C
+- [ ] ASIL-D
+
+What levels of IEC-61508 do you work with?
+
+- [ ] SIL-1
+- [ ] SIL-2
+- [ ] SIL-3
+- [ ] SIL-4
+
+What levels of DO-178 do you work with?
+
+- [ ] DAL A
+- [ ] DAL B
+- [ ] DAL C
+- [ ] DAL D
+- [ ] DAL E
+
+What levels of IEC-62304 do you work with?
+
+- [ ] Class A
+- [ ] Class B
+- [ ] Class C
+
+What coding guidelines do you work with in your safety critical work?
+*Select multiple
+
+- [ ] MISRA
+- [ ] AEC
+- [ ] Other (text entry)
```

Comment:
Here are some other suggestions for coding guidelines,

- AUTOSAR (C++)
- JSF AV (C++)
- CERT (C, C++, Java)
- HIC++
- CWE
    - Not truly a set of guidelines or standards, but may be considered part of the category